src/pybind/mgr/prometheus/module.py

```diff
@@ -156,6 +157,7 @@ class Module(MgrModule):
             {'name': 'server_addr'},
             {'name': 'server_port'},
             {'name': 'scrape_interval'},
+            {'name': 'rbd_stats_pools'},
```

Comment:
Nit: can you add a "max result" limit optional and default it to something semi-sane (like top 20 or 50)? If someone overrides it, at least it's not configured out-of-the-box to grab too much data.

Reply:
I am not sure here. I think I would agree to have some safety limit, just in case it will really produce too much in some deployments (similarly how we have limits like `osd_max_object_size`), but it should be much  higher then 50. Considering that one entry size would be about 80 bytes, I think we could set the limit to 16K (about 1M per message) or even 64K (about 5M per message). I suppose the installations that may hit such a limit (i.e having about 100K active images?) can afford report messages of such a size?

Setting it just to 20 or 50 does not make much sense to me, because it would mean that almost everyone would hit it, and would have incomplete stats, difficult to interpret.

Reply:
Well, it would provide the top N image stats per OSD (note that I still want to use weighted probability to select that top N-ish results at the OSD side in a future PR), not only N images in the result.   I'd be interested to see a test where Prometheus was ingesting 5MiB of stats every 5 seconds (or so) and see if it can keep up with the load. That's why I think it's safer to start small.

Reply:
Exactly, if you expect the prometheus will have problems earlier with such amount of data, why we need to limit osds here? It will not help for prometheus (because it still will receive stats for all images in pool), but in most cases with such default settings (top 20 or 50) the stats will be inaccurate. And it will be not so easy for a user to realize what is wrong here.

By default it is disabled. Users should explicitly specify the pool(s) (or just a namespace, I will add the support for this later) they want to monitor. If it causes an issue they can just disable it. Having a high enough limit (16K) is probably good as a safety guard.

I will try to to stress test this (and prometheus) with a large number of images.

Reply:
Sorry -- it was two separate thoughts in there (noting that limiting the stats would be top-N per OSD *and* the query if Prometheus can really handle a large ingestion of data like that).